vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/ReasonerImpl.java

```diff
@@ -139,7 +140,7 @@ public void load() throws AlreadyStartedException, EDBConfigurationException, IO
 			edbPredicates.add(edbPredConfig.getPredicate());
 		}
 		for (final Atom fact : this.facts) {
-			edbPredicates.add(fact.getPredicate());
+			edbPredicates.add(fact.getPredicate().getName());
```

Comment:
Here and below in this file: Shouldn't we just use Predicate objects here instead of strings?

Reply:
I will fix this in the "query" branch